platform/api/packet/api.go

```diff
@@ -478,22 +478,46 @@ boot`, a.opts.InstallerImageKernelURL, userdataURL, linuxConsole[a.opts.Board],
 }
 
 // device creation seems a bit flaky, so try a few times
-func (a *API) createDevice(hostname, ipxeScriptURL string) (device *packngo.Device, err error) {
+func (a *API) createDevice(hostname, ipxeScriptURL, id string) (*packngo.Device, error) {
 	for tries := apiRetries; tries >= 0; tries-- {
-		var response *packngo.Response
-		device, response, err = a.c.Devices.Create(&packngo.DeviceCreateRequest{
-			ProjectID:     a.opts.Project,
-			Facility:      []string{a.opts.Facility},
-			Plan:          a.opts.Plan,
-			BillingCycle:  "hourly",
-			Hostname:      hostname,
-			OS:            "custom_ipxe",
-			IPXEScriptURL: ipxeScriptURL,
-			Tags:          []string{"mantle"},
-		})
+		var (
+			device   *packngo.Device
+			response *packngo.Response
+			err      error
+		)
+
+		if id != "" {
+			plog.Infof("Recycling instance: %s", id)
+			// we update the device and we force a PXE boot in order to fetch the
+			// new configuration.
+			alwaysPXE := true
+			device, response, err = a.c.Devices.Update(id, &packngo.DeviceUpdateRequest{
```

Comment:
we have to device what to do when the update fails, currently it deletes the device and creates a new one, this is not wanted but instead we want to retry calling this function

Reply:
(currently the error is discarded but even if not this is a problem)

Reply:
An other alternative is to simplify this logic and directly terminate here without retry :)